test/nodes/test_llm_openai.py

```
@@ -0,0 +1,402 @@
+        ig = self._make_iglobal(config, mock_config)
+
+        fake_client = _FakeOpenAI()
+        fake_client.chat.completions.create = MagicMock(side_effect=_FakeAPIConnectionError('Connection error'))
+        with patch('openai.OpenAI', return_value=fake_client):
+            ig.validateConfig()
+
+        assert len(warned_messages) == 1
+        assert 'Connection error' in warned_messages[0]
+
+    @pytest.mark.parametrize('model', ['gpt-5', 'gpt-5.1', 'gpt-5-mini', 'gpt-5-nano'])
+    def test_newer_models_use_max_completion_tokens(self, model, mock_config, warned_messages):
```

Ruff (0.15.7)

[warning] Unused method argument: `warned_messages`

(ARG002)